### ISSUE TITLE
treemacs-collapsed-dirs is broken with CJK characters

### ISSUE THREAD

**Issue by y-usuzumi** [Alexander-Miller/treemacs#565]
There are several related issues. See the screenshot below:

![无标题](https://user-images.githubusercontent.com/1560189/68560329-9aa1b980-047b-11ea-8df9-8106ac12ef19.png)

Bad 1: When a file/directory name contains CJK characters and should collapse into the parent name, it is not displayed correctly. I can't expand the directory either:

![image](https://user-images.githubusercontent.com/1560189/68560649-194b2680-047d-11ea-9565-97f8eadbf0bf.png)

Bad 2: When a parent directory contains CJK characters, the single child is never collapsed.
Bad 3: Bad 2 in a recursive manner: A parent directory containing CJK characters will stop any further collapse in its children.

I personally prefer that it is never collapsed, but the configuration variable as demonstrated in spacemacs doc does not work for me:

```
(setq-default dotspacemacs-configuration-layers '(
  (treemacs :variables treemacs-use-collapsed-directories 3)))
```

I tried setting it to zero and nothing happened. Since I hardly know any emacs lisp I can't help debugging in a serious way but I boldly guess this has something to do with treemacs-async.el overwriting the value. It should belong to a separate issue anyway.

**Comment by Alexander-Miller** [comment]
> I tried setting it to zero and nothing happened.

You're right that the value is overwritten. Spacemacs should set it /after/ treemacs is loaded, not before. I'll open a PR for it. In the meantime you can put this into your `dotspacemacs/user-init`:

```
(with-eval-after-load 'treemacs
  (setf treemacs-collapse-dirs 0))
```
(That variable may stop working in a couple moths' time because it'll be deprecated and replaced with `treemacs-flatten-dirs` which is a more appropriate name and doesn't clash withe the open/expand close/collapse terminology)

> When a file/directory name contains CJK characters and should collapse into the parent name, it is not displayed correctly


My guess is that the issue is caused by some incorrect conversion or encoding issue, since the flattening info is fetched asynchronously in python. Can you post an example name that does not work? I'll see if I can reproduce it on my system.

**Comment by y-usuzumi** [comment]
Thank you!

Please try any of this (Chinese and Japanese respectively):

```
你好
ありがとう
```

**Comment by Alexander-Miller** [comment]
https://github.com/syl20bnr/spacemacs/pull/12960

**Comment by Alexander-Miller** [comment]
https://github.com/Alexander-Miller/pfuture/issues/6 should take care of this as well.

**Comment by y-usuzumi** [comment]
Hi. I can confirm the pfuture fix works.

However, after upgrading both treemacs and pfuture treemacs-collapse-dirs stops working at all.
Now nothing is collapsed.

**Comment by Alexander-Miller** [comment]
What does the following code output?

```Elisp
(pfuture-await-to-finish
 (pfuture-new
  "python"
  "treemacs-dir/treemacs-dirs-to-collapse.py"
  "your-project-dir"
  "3" "t"))
```

**Comment by y-usuzumi** [comment]
It seems whatever I put at `your-project-dir` I get an empty string.

**Comment by y-usuzumi** [comment]
I got this:
```
"((\"/b/c\" \"/home/kj/Lab/Netis/特殊任务/2019-12-30 DBCube补丁/a\" \"/home/kj/Lab/Netis/特殊任务/2019-12-30 DBCube补丁/a/b\" \"/home/kj/Lab/Netis/特殊任务/2019-12-30 DBCube补丁/a/b/c\"))"
```

**Comment by Alexander-Miller** [comment]
Doesn't look wrong, assuming the filenames are right, my system cannot display every character.

If you call `read` on that string you should receive a proper elisp list.

Let's see if it's a more general problem. Does this work:

```Èlisp
(with-current-buffer (treemacs-get-local-buffer)
  (treemacs-goto-file-node
   "/home/kj/Lab/Netis/特殊任务/2019-12-30 DBCube补丁/a"))
```

Then set `treemacs-collapse-dirs` to 0 and refresh your project and run this:
```Elisp
(with-current-buffer (treemacs-get-local-buffer)
  (let ((treemacs-collapse-dirs 3))
    (treemacs--collapse-dirs
     (treemacs--parse-collapsed-dirs
      (pfuture-new
       "python"
       "treemacs-dir/treemacs-dirs-to-collapse.py"
       "your-project-dir"
       "3" "t")))))
```
Does that change anything or produce any error?

**Comment by y-usuzumi** [comment]
Hi, I got the following error when running the first code snippet:

```
Debugger entered--Lisp error: (wrong-type-argument integer-or-marker-p nil)
  treemacs-find-file-node("/home/kj/Lab/Netis/特殊任务/2019-12-30 DBCube补丁/a" nil)
  treemacs-goto-file-node("/home/kj/Lab/Netis/特殊任务/2019-12-30 DBCube补丁/a")
  (save-current-buffer (set-buffer (treemacs-get-local-buffer)) (treemacs-goto-file-node "/home/kj/Lab/Netis/特殊任务/2019-12-30 DBCube补丁/a"))
  eval((save-current-buffer (set-buffer (treemacs-get-local-buffer)) (treemacs-goto-file-node "/home/kj/Lab/Netis/特殊任务/2019-12-30 DBCube补丁/a")) nil)
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

Sorry I'm still very noob at Emacs Lisp. Not sure what else I can do with this error.

**Comment by Alexander-Miller** [comment]
It works on my end now that I have the right pfuture version: https://imgur.com/z2iM7BG Make sure that this changed is included in the one you have installed: https://github.com/Alexander-Miller/pfuture/commit/fbecd1562b34fcd15563b7cc00d98e5af0d7e43d
`

**Comment by y-usuzumi** [comment]
It works now!

I remembered having checked the patch was included yesterday (pfuture version 1.7) when I found it did not work. But sometime later I cleaned all .elc files. Maybe that's when the patch came into effect.

Thank you very much!
